samples: drivers: adc: add demo and fix STM32 sequence configuration (#25590)
zephyrproject-rtos/zephyr: samples/drivers/adc/README.rst

.. _adc-sample:

Analog-to-Digital Converter (ADC)
#################################

Overview
********

This sample demonstrates how to use the ADC driver API.

Depending on the MCU type, it reads the ADC samples of one or two ADC channels
and prints the raw readings to the console.

The pins of the ADC channels are board-specific. Please refer to the board
or MCU datasheet for further details.

The sample should work with most MCU series, including STM32, NXP Kinetis and

> Microchip SAM0. Nordic nRFx MCUs are currently not supported, as they require
> additional configuration for their differential input feature.

Reviewer: ```suggestion
Microchip SAM0. Nordic nRFx MCUs are currently not supported.
```

The stated  reason why is not totally accurate, so better to just say they're not supported IMO.

Building and Running
********************

The ADC peripheral and pinmux is configured in the board's ``.dts`` file. Make
sure that the ADC is enabled (``status = "okay";``).

> In addition to that, the ADC channel has to be specified in the ``io-channels``

Reviewer: ```suggestion
In addition to that, this sample requires an ADC channel specified in the ``io-channels``
```

Otherwise users might think this is a general requirement of the API, I think.
